### PR TITLE
Made the library Asyncronous

### PR DESCRIPTION
**Pull request by cauefcr**
I added two channels to the project, a Go one and a C one ([eb_chan](https://github.com/davekeck/eb_chan)), the C one is accessed from the dispatch_process, where the events are sent in json form, the second one is accessible to the user, is used by the pollEv function. pollEv reads from the C channel, converts the json to the Event struct and sends it to the Go channel.

The project only has two functions now: Start and End, which attaches and detaches the hook, as well as manage the global state, respectively.

(hope the PR is accepted :) )

### CODE REVIEW

**Comment by cauefcr** [comment]
What do you think about adding routing support, so the library user can do something like:
`Go
hook.Pressed("/kbd/a",func(e){
    fmt.Print("a was pressed")
})
`

File: hook.go [hunk 1]
```diff
@@ -19,28 +19,92 @@ package hook
...
 import "C"
 
 import (
-	// 	"fmt"
-	"unsafe"
+	"time"
 )
 
-// AddEvent add event listener
-func AddEvent(key string) int {
```
Comment (vcaesar) [inline hook.go]: Please keep the compatibility of the original code.

Comment (cauefcr) [reply hook.go]: Do you mean that it should check if the key is pressed at that exact moment? the previous behavior was blocking until the key is pressed, but i'm not so sure that is desirable.

What about adding a callback to the AddEvent function?

Comment (vcaesar) [reply hook.go]: I mean don't break the compatibility of the api, if it's new features keep the original code.

**Comment by vcaesar** [comment]
The code has been tested on one platform, and will be further reviewed and tested on other platforms to merge.

**Comment by vcaesar** [comment]
Thanks, this is awesome work.

**Comment by cauefcr** [comment]
You're quite welcome, thanks for robotgo.

**Pull request merged**